docs/troubleshooting.md

```diff
@@ -0,0 +1,333 @@
+```sh
+# Expect your Onload CR(s)
+kubectl get onload
+
+# Expect Onload Device Plugin DaemonSet, 'onload-module' Module, and optionally 'onload-sfcmod' Module
+kubectl get ds,pod,module -l app.kubernetes.io/managed-by=onload-operator
+
+# Expect Status of Module Loader to be Available
+kubectl describe module onload-module
+
+# Optional
+kubectl describe module onload-sfcmod
+```
+
+#### Onload Device Plugin containers
+
+```sh
+kubectl describe pod -l app.kubernetes.io/component=device-plugin
+
+# Expect files copied by init container
+kubectl logs -l app.kubernetes.io/component=device-plugin -c init -f
+
+# Expect Onload Worker (containing Onload Control Plane) to have loaded
+kubectl logs -l app.kubernetes.io/component=device-plugin -c onload-worker --tail 200 -f
+
+# Expect listing of provided mounts, discovered interfaces, and RPC server up
+kubectl logs -l app.kubernetes.io/component=device-plugin -c device-plugin -f
+```
+
+### Registries
+
+Are the images accessible to the cluster? Either by allowing access the Internet sources (DockerHub) or
+providing and configuring container images for an internal registry.
+
+If using in-cluster builds, has a container image registry with write access been configured within your cluster?
+
+Has the registry/ies been configured in the cluster, Service Account, and/or Onload CR as required, such as
+being insecure, internally-signed, or with a pull secret. In OpenShift,
+this is configured in `image.config.openshift.io/cluster` CR.
+
+Are the images stale? This is particularly of concern if images have been cached locally using tags that do not
+completely uniquely identify the image, including any particular build parameters if they are built in-house. Related
+to this is the `imagePullPolicy` in Onload CR.
+
+### OpenShift Image Registry
+
+KMM and Onload Operator will work with any writable registry, however this repo's OCP-specific Onload CR samples
+utilise the OpenShift Image Registry by default to store in-cluster build images.
+
+```sh
+# Expect configured, eg. spec.managementState, spec.defaultRoute, and spec.storage
+oc get configs.imageregistry.operator.openshift.io/cluster -o yaml
+
+# Expect `onload-module` when using for kernel module in-cluster builds
+oc get imagestreams -n onload-clusterlocal
+
+# Expect list of KMM-built or pre-built images
+oc describe imagestream onload-module -n onload-clusterlocal
+```
+
+### KMM Operator version
+
+```sh
+# When deployed via OLM (OperatorHub), version:
+kubectl get csv -n openshift-kmm
+
+# else, version in labels and image tags:
+kubectl describe -n openshift-kmm deployment/kmm-operator-controller-manager
+```
+
+### KMM Operator operation
+
+```sh
+# Expect events: ModuleLoaded, ModuleUnloaded
+kubectl describe node $node
+kubectl get event -n openshift-kmm
+
+kubectl logs -n openshift-kmm deployment/kmm-operator-controller-manager -f
+```
+
+Refer to KMM documentation for further expected deployment state details and troubleshooting advice.
```

Comment:
It might be worth highlighting the fact that there may be issues with the module deployment that are not visible by the onload operator or by examining the modules, and so the actual kmm pods have to be examined manually. I know you include the command to get the logs from the kmm controller, but I think a small explanation why would be helpful.